### PR TITLE
Upgrade to go 1.20

### PR DESCRIPTION
**Pull request by johnsimons**
Added support for arm building.
Updated dependencies to latest

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+	var debugMode bool
+	flag.BoolVar(&debugMode, "debug", false, "set to true to run the provider with support for debuggers like delve")
+	flag.Parse()
+
+	opts := &plugin.ServeOpts{
+		ProviderFunc: octopusdeploy.Provider}
+
+	if debugMode {
+		opts.Debug = true
+		opts.ProviderAddr = "octopus.com/com/octopusdeploy"
```
Comment (jbristowe) [inline main.go:21]: My guess is that this _might_ fail if the provider is pulled from the Terraform Registry -- it uses a different address.

**Review by jbristowe — COMMENTED** [review]
Unsure about debug address (see comment)

File: main.go (hunk 1 above)
Comment (johnsimons) [reply main.go:21]: I follow the instructions in https://opencredo.com/blogs/running-a-terraform-provider-with-a-debugger/.
And it works 😄 

So @jbristowe, this is only the address to be used when we start with `--debug` argument, which would be done using the local copy.

I hope this makes sense, let me know?

Comment (johnsimons) [reply main.go:21]: In other words, it wouldn't make sense to try to attach the debugger to a non-local compiled version.

**Review by jbristowe — APPROVED** [review]

**Pull request merged**